chain/client/src/client.rs

```diff
@@ -58,6 +59,7 @@ use near_primitives::version::PROTOCOL_VERSION;
 use near_primitives::views::{CatchupStatusView, DroppedReason};
 
 const NUM_REBROADCAST_BLOCKS: usize = 30;
+const CHUNK_HEADERS_FOR_INCLUSION_CACHE_SIZE: usize = 100;
```

Comment:
Is this number big enough? How do you make sure recent blocks are not purged from this list?

Reply:
Well this is 100 blocks. It seems hard to think of a scenario where a block producer would first hear of a chunk, and then hear of new chunks for 100 other blocks (all already within horizon as validated by the ShardsManager) before it is supposed to use this chunk for producing a block...

Reply:
actually I realized that the horizon we used to use for headers-ready-to-be-included was only 10, but the horizon we used for storing blocks is 1024 (behind) + 5 (ahead). So... since we're now using the second horizon for notifying the client for ready-to-include chunks, I bumped it to 2048 - enough for every single block in horizon to be notified out of order plus a lot of room.